site/plugins.md

```diff
@@ -390,8 +391,245 @@ func (p *Plugin) Report(w http.ResponseWriter, r *http.Request) {
 
 ```
 
-**See Also**
+### <a id="report-data-structures"></a>Report Data structures
+A report can contain many types of information.
+If you go back to the [Reporter Interface](#reporter-interface) section, you see the top-level `Plugins` attribute.
+Along with that, a report may contain multiple topologies.
+An example of a report containing few topologies is the following:
 
-  * [Building Scope](/site/building.md)
+```json
+{
+  "Host": {},
+  "Container": {},
+  "Process": {},
+  ...,
+  "Plugins": [...,]
+}
+```
+
+### Topologies
+Topologies can be of various types.
+Each topology consists of a list of nodes and other information about the topology.
+The available topologies are:
+
+- `Endpoint` nodes are (address, port) tuples on each host.
+- `Process` nodes are processes on each host.
+- `Container` nodes represent all Docker containers on hosts running probes.
+- `Pods` nodes represent all Kubernetes pods running on hosts running probes.
+- `Service` nodes represent all Kubernetes services running on hosts running probes.
+- `Deployment` nodes represent all Kubernetes deployments running on hosts running probes.
+- `ReplicaSet` nodes represent all Kubernetes ReplicaSets running on hosts running probes.
+- `ContainerImage` nodes represent all Docker container images on hosts running probes.
+- `Host` nodes are physical hosts that run probes.
+- `Overlay` nodes are active peers in any software-defined network that's overlaid on the infrastructure.
+
+The topology structure consists of the following attributes:
+
+- `nodes` - is the list of the nodes that compose the topology.
+- `controls` - contains the list of IDs of the active controls at a particular time.
+- `metadata_templates` - contains the templates used to render data into the Scope UI.
+- `table_templates` - contains the templates used to render tables into the Scope UI.
+- `metric_templates` - contains the templates used to render metrics into the Scope UI.
+
+**Note**: These attribute are not required. But a topology with no `nodes` does not have any information to render, `metadata_templates`, as well as `table_templates`, are needed to know how to render the information carried by `nodes` in the Scope UI.
+
+### Nodes
+A Node contains information about a specific element of a topology.
+For example the Host topology will contain nodes describing all the hosts in it.
+The same is for containers and Container topology, pods and Pod topology and so on.
```

Comment:
The same applies for containers and the Container topology, pods and the Pod topology and so on